packages/compass-explain-plan/src/components/explain-tree/explain-tree-stage.tsx

```diff
@@ -60,6 +61,70 @@ interface ExecutionstatsProps {
   totalExecTimeMS: number;
 }
 
+// Instead of using CSS ellipsis which trims the text towards the end / front,
+// we use this custom implementation to trim the text in middle because shard
+// names could have similar text in the front and the differentiating numbers
+// are generally available at the end of the name. For example: atlas-shard-0,
+// atlas-shard-1, etc. The trimThreshold is taken from a hit and try approach to
+// fit as many big chars as possible (worst case scenario) in the shard card
+// because other sophisticated approaches require understanding font specifics
+// like kerning, font-constant, etc to properly determine the width of character
+// for a particular font.
+export const trimInMiddle = (
+  text: string,
+  trimThreshold = 20,
+  charsToKeepInFront = 6,
+  charsToKeepInBack = 4
+) => {
+  if (text.length <= trimThreshold) {
+    return text;
+  }
+
+  const charsBeforeEllipsis = text.substring(0, charsToKeepInFront);
+  const remainingText = text.substring(charsToKeepInFront);
+  const charsAfterEllipsis = remainingText.substring(
+    remainingText.length - charsToKeepInBack
+  );
+  return `${charsBeforeEllipsis}...${charsAfterEllipsis}`;
```

Comment:
```suggestion
  return `${charsBeforeEllipsis}…${charsAfterEllipsis}`;
```

Reply:
That's interesting. What is that character? 🧐

Reply:
[Horizontal ellipsis](https://www.compart.com/en/unicode/U+2026). It's nice because it's literally one character instead of three, so more space saving 😄